src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -2304,41 +2295,11 @@ private unsafe void ReplaceInPlaceAtChunk(ref StringBuilder? chunk, ref int inde
         /// This method prevents out-of-bounds writes in the case a different thread updates a field in the builder just before a copy begins.
         /// All interesting variables are copied out of the heap into the parameters of this method, and then bounds checks are run.
         /// </remarks>
-        private static unsafe void ThreadSafeCopy(char* sourcePtr, char[] destination, int destinationIndex, int count)
-        {
-            if (count > 0)
-            {
-                if ((uint)destinationIndex <= (uint)destination.Length && (destinationIndex + count) <= destination.Length)
-                {
-                    fixed (char* destinationPtr = &destination[destinationIndex])
-                        string.wstrcpy(destinationPtr, sourcePtr, count);
-                }
-                else
-                {
-                    throw new ArgumentOutOfRangeException(nameof(destinationIndex), SR.ArgumentOutOfRange_Index);
-                }
-            }
-        }
-
-        private static unsafe void ThreadSafeCopy(char[] source, int sourceIndex, Span<char> destination, int destinationIndex, int count)
-        {
-            if (count > 0)
-            {
-                if ((uint)sourceIndex > (uint)source.Length || count > source.Length - sourceIndex)
-                {
-                    throw new ArgumentOutOfRangeException(nameof(sourceIndex), SR.ArgumentOutOfRange_Index);
-                }
-
-                if ((uint)destinationIndex > (uint)destination.Length || count > destination.Length - destinationIndex)
-                {
-                    throw new ArgumentOutOfRangeException(nameof(destinationIndex), SR.ArgumentOutOfRange_Index);
-                }
+        private static unsafe void ThreadSafeCopy(char* sourcePtr, char[] destination, int destinationIndex, int count) =>
+            new ReadOnlySpan<char>(sourcePtr, count).CopyTo(new Span<char>(destination, destinationIndex, count));
 
-                fixed (char* sourcePtr = &source[sourceIndex])
-                fixed (char* destinationPtr = &MemoryMarshal.GetReference(destination))
-                    string.wstrcpy(destinationPtr + destinationIndex, sourcePtr, count);
-            }
-        }
+        private static void ThreadSafeCopy(char[] source, int sourceIndex, Span<char> destination, int destinationIndex, int count) =>
```

Comment:
Now that it is this simple, I am wondering whether we need these methods at all. We can gain minor additional efficiencies by just inlining it everywhere: 
- Most places pass in `char[]` as the destination, so we pay at the `char[]`->`Span` conversion at the callsite just to slice the Span again here. It is better to just slice the array as part of the conversion.
- Some places pass zero for the indices,